File: 125-valid-palindrome/125-valid-palindrome.py
import re

class Solution:
    def isPalindrome(self, s: str) -> bool:
        # Cleaning input string
        s = re.sub('[^a-zA-Z0-9]+', '',s).lower()
        left = 0
        right = len(s) - 1
        
        while (left < right):
            if (s[left] != s[right]): return False
            left = left + 1
            right = right - 1
        return True
        

File: 125-valid-palindrome/125-valid-palindrome.rs
impl Solution {
    // O(n) time | O(1) space
    pub fn is_palindrome(s: String) -> bool {
        // Cleaning input string
        let cleaned_string = s
            .chars() // returns iterator over the chars of string slice
            .filter(|c| c.is_alphanumeric()) // returns iterator over filtered string slice
            .map(|c| c.to_ascii_lowercase()); // maps each char to lowercase
        cleaned_string.clone().eq(cleaned_string.rev())
    }
}